src/megatron/energon/media/extractor.py

```diff
@@ -0,0 +1,215 @@
+# Copyright (c) 2025, NVIDIA CORPORATION.
+# SPDX-License-Identifier: BSD-3-Clause
+
+from __future__ import annotations
+
+import io
+import logging
+from dataclasses import dataclass
+from enum import Enum
+from fnmatch import fnmatch
+from pathlib import Path
+from typing import Union
+
+import filetype
+from PIL import Image, UnidentifiedImageError
+
+from megatron.energon.av import AVDecoder
+from megatron.energon.media.metadata import ImageMetadata, MediaMetadataBase, MediaMetadataType
+
+logger = logging.getLogger(__name__)
+
+
+SourceData = Union[bytes, Path, io.IOBase]
+
+
+class MediaFilterStrategy(str, Enum):
+    """Strategy used to decide whether an entry should be treated as media."""
+
+    EXT = "EXT"  # by file extension
+    TYPE = "TYPE"  # by filetype (detected using filetype package)
+    PATTERN = "PATTERN"  # by glob pattern (e.g. '*.jpg')
+
+
+@dataclass(frozen=True)
+class MediaFilterConfig:
+    """Configuration for media detection during dataset preparation."""
+
+    strategy: MediaFilterStrategy
+    pattern: str | None = None
+
+    @classmethod
+    def parse(cls, value: str | None) -> "MediaFilterConfig":
+        if value is None:
+            value = "EXT"
+
```

Comment:
```suggestion
    def parse(cls, value: str = "EXT") -> "MediaFilterConfig":

```
Also, see comment on naming "EXT"